make/common/native/Flags.gmk

```diff
@@ -63,6 +63,11 @@ define SetupCompileFileFlags
     endif
   endif
 
+  ifeq ($$($1_LINK_TIME_OPTIMIZATION), true)
+    $1_OPT_CFLAGS += $(C_O_FLAG_LTO)
+    $1_OPT_CXXFLAGS += $(CXX_O_FLAG_LTO)
+  endif
```

Comment:
I don't think adding the LTO flags to the OPT flags is the right move, but if we are going with that, then this addition of LTO flags should only be done if `$$($1_OPTIMIZATION)` is set, as otherwise, those flags are already added through `$$($2_OPT_CFLAGS)`. That means, this block should be moved into the else block above, before the endif on line 64. 

However, the OPT flags are treated separately through SetupCompilerFlags and SetupCompileFileFlags because it should be possible to override the optimization level on a per file basis. The LTO flags on the other hand, are not possible to override on a per file basis, so we should not be tinkering with those in the SetupCompileFileFlags macro at all. So mixing in the LTO flags with the OPT flags is the wrong move.

Reply:
Again, the optimization flags aren't handled as a separate set of variables because they are of the category "optimization". It's because optimization flags are handled on a higher level with an abstraction of "low", "medium", "high" etc, and that abstraction allows for specific file overrides. That's what the `*_OPT_C*FLAGS` are there to handle. Since the LTO flags are not related to this abstraction, they should not be mixed into these variables, that is just confusing and adding unnecessary complexity.

The LTO flags should just be stacked on to the `$1_EXTRA_*` flags like any other.

Reply:
I adjusted my changes to Flags.gmk .